File: Perl/Endpoint Examples/Multipart Payload/blank-pdf.pl
#!/usr/bin/env perl
use strict;
use warnings;
use utf8;
use FindBin qw($Bin);
use LWP::UserAgent;
use HTTP::Request::Common qw(POST);
use Dotenv;

#!
# What this sample does:
# - Calls /blank-pdf with multipart/form-data to create a three-page blank PDF.
#
# Setup (.env):
# - Copy .env.example to .env (Perl folder root)
# - Set PDFREST_API_KEY=your_api_key_here
# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
#     PDFREST_URL=https://eu-api.pdfrest.com
# For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
#
# Usage:
#   perl "Endpoint Examples/Multipart Payload/blank-pdf.pl"
#
# Output:
# - Prints the API JSON response to stdout. Non-2xx responses exit with a concise message.

binmode STDOUT, ':raw';
binmode STDERR, ':encoding(UTF-8)';

my $env_path = "$Bin/../../.env";
-e $env_path and Dotenv->load($env_path);

my $api_key = $ENV{PDFREST_API_KEY} // '';
if (!$api_key || $api_key =~ /^\s*$/) {
    print STDERR "Missing PDFREST_API_KEY in .env or environment\n";
    exit 1;
}

my $api_base = $ENV{PDFREST_URL} // $ENV{PDFREST_API} // 'https://api.pdfrest.com';
$api_base =~ s{/+$}{};

my $ua = LWP::UserAgent->new( timeout => 60 );

eval {
    my $req = POST(
        "$api_base/blank-pdf",
        'Content_Type' => 'form-data',
        'Content'      => [
            page_size  => 'letter',
            page_count => '3',
            page_orientation => 'portrait',
        ]
    );
    $req->header('api-key' => $api_key);

    my $resp = $ua->request($req);
    print STDOUT $resp->decoded_content // '';
    if (!$resp->is_success) {
        print STDERR "\nblank-pdf request failed with status " . $resp->code . "\n";
        exit 1;
    }
    1;
} or do {
    my $err = $@ || 'Unknown error';
    $err =~ s/\s+$//;
    print STDERR "Error: $err\n";
    exit 1;
};

__END__

File: Ruby/Endpoint Examples/Multipart Payload/blank-pdf.rb
#!
# What this sample does:
# - Calls /blank-pdf via multipart/form-data to create a blank three-page PDF.
#
# Setup (.env):
# - Copy .env.example to .env
# - Set PDFREST_API_KEY=your_api_key_here
# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
#     PDFREST_URL=https://eu-api.pdfrest.com
#   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
#
# Usage:
#   ruby "Endpoint Examples/Multipart Payload/blank-pdf.rb"
#
# Output:
# - Prints the API JSON response to stdout. Non-2xx responses abort with a concise message.

require "faraday"
require "faraday/retry"
require "faraday/multipart"
require "dotenv"

Dotenv.load

API_KEY = ENV["PDFREST_API_KEY"]
abort("Missing PDFREST_API_KEY in .env") if API_KEY.nil? || API_KEY.strip.empty?

API_BASE = (ENV["PDFREST_URL"] || ENV["PDFREST_API"] || "https://api.pdfrest.com").sub(%r{/+$}, "")

conn = Faraday.new(url: API_BASE) do |f|
  f.request :retry, max: 2, interval: 0.2
  f.request :multipart
  f.adapter Faraday.default_adapter
end

body = {
  page_size: "letter",
  page_count: "3",
  page_orientation: "portrait"
}

response = conn.post("/blank-pdf") do |req|
  req.headers["api-key"] = API_KEY
  req.body = body
end

puts response.body
abort("blank-pdf request failed with status #{response.status}") unless response.success?
